src/org/traccar/notification/NotificationMail.java

```diff
@@ -87,9 +87,9 @@ private static Properties getProperties(PropertiesProvider provider) {
     public static void sendMailSync(long userId, Event event, Position position) throws MessagingException {
         User user = Context.getPermissionsManager().getUser(userId);
 
-        Properties properties = getProperties(new PropertiesProvider(Context.getConfig()));
-        if (!properties.containsKey("mail.smtp.host")) {
-            properties = getProperties(new PropertiesProvider(user));
+        Properties properties = getProperties(new PropertiesProvider(user));
+        if (Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig") || !properties.containsKey("mail.smtp.host")) {
```

Comment:
Seems a bit inefficient to initialise user properties if `ignoreUserConfig` is set to `true`.

Reply:
I also noticed it, but it become look more complex
```java
        Properties properties;
        if (!Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig")) {
            properties = getProperties(new PropertiesProvider(user));
            if (!properties.containsKey("mail.smtp.host")) {
                properties = getProperties(new PropertiesProvider(Context.getConfig()));
            }
        } else {
            properties = getProperties(new PropertiesProvider(Context.getConfig()));
        }
        if (!properties.containsKey("mail.smtp.host")) {
            Log.warning("No SMTP configuration found");
            return;
        }
```
is it OK? or you see how make simpler?

Reply:
How about something like this:
```java
Properties properties = null;
if (!Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig")) {
    properties = getProperties(new PropertiesProvider(user));
}
if (properties == null || !properties.containsKey("mail.smtp.host")) {
    properties = getProperties(new PropertiesProvider(Context.getConfig()));
}
```